test/base/ReferenceCounterTest.ooc

```diff
@@ -1,11 +1,69 @@
+use ooc-unit
 use ooc-base
+import threading/Thread
 
-Thing: class {
-	init: func
+TestObject: class {
+	_isAlive: Bool*
+	init: func (=_isAlive) {
+		this _isAlive@ = true
+	}
+	free: override func {
+		this _isAlive@ = false
+		super()
+	}
 }
 
-t := Thing new()
-r := ReferenceCounter new(t)
-r update(-1)
+ReferenceCounterTest: class extends Fixture {
+	init: func {
+		super("ReferenceCounter")
+		this add("single thread", func {
+			isAlive: Bool
+			object := TestObject new(isAlive&)
+			expect(isAlive)
+			counter := ReferenceCounter new(object)
+			for (i in 0 .. 10)
+				counter increase()
+			for (i in 0 .. 10)
+				counter decrease()
+			expect(isAlive, is false)
+			counter free()
+		})
+		this add("multiple threads", This _testMultipleThreads)
+	}
+	_testMultipleThreads: static func {
+		isAlive: Bool
+		object := TestObject new(isAlive&)
+		counter := ReferenceCounter new(object)
+		numberOfThreads := 8
+		countPerThread := 500
+		threads := Thread[numberOfThreads] new()
+		expect(isAlive)
+		job := func {
+			for (i in 0 .. countPerThread) {
+				counter increase()
+				Thread yield()
+				counter decrease()
+				Thread yield()
+			}
+		}
+		counter increase()
+		for (i in 0 .. numberOfThreads) {
+			threads[i] = Thread new(|| job())
```

Comment:
Yes, that's why I pointed it out. `|| job()` will create a closure that calls `job()` and that's all it does. So why not just give the closure `job` to the thread instead of wrapping it inside another closure? Is it because you can't, because it doesn't work?